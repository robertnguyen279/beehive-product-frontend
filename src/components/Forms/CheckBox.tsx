import React from 'react';
import { v4 as uuid } from 'uuid';
import { CheckBoxType } from 'types/main';

const CheckBox = ({ name, value, label, className, onChange }: CheckBoxType): React.ReactElement => {
  const id = name + uuid();
  return (
    <div className={`form__checkbox flex items-center ${className ? className : ''}`}>
      <input type="checkbox" name={name} value={value} id={id} className="z-10" onChange={onChange} />
      <span className="form__checkbox__checkmark">.</span>
      <label htmlFor={id} className="pl-3 cursor-pointer text-sm text-gray-600">
        {label}
      </label>
    </div>
  );
};

export default CheckBox;
