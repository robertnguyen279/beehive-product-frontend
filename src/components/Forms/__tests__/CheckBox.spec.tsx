import React from 'react';
import { shallow } from 'enzyme';

import CheckBox from '../CheckBox';
// const clickFn = jest.fn();

describe('CheckBox', () => {
  const component = shallow(
    <CheckBox
      name="test"
      label="test"
      value="test"
      onChange={(e) => {
        console.log(e);
      }}
    />,
  );

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
