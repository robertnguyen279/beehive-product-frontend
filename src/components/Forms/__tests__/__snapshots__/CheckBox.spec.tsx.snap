// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should render correctly 1`] = `
<div
  className="form__checkbox flex items-center "
>
  <input
    className="z-10"
    id="test"
    name="test"
    onChange={[Function]}
    type="checkbox"
    value="test"
  />
  <span
    className="form__checkbox__checkmark"
  >
    .
  </span>
  <label
    className="pl-3 cursor-pointer text-sm text-gray-600"
    htmlFor="test"
  >
    test
  </label>
</div>
`;
